src/main/scala/wdlTools/syntax/package.scala

```diff
@@ -25,24 +25,47 @@ object WdlVersion {
 // source location in a WDL program. We add it to each syntax element
 // so we could do accurate error reporting.
 //
+// Note: 'line' and 'col' may be 0 for "implicit" elements. Currently,
+// the only example of this is Version, which in draft-2 documents has
+// an implicit value of WdlVersion.Draft_2, but there is no actual version
+// statement.
+//
 // line: line number
 // col : column number
 // URL:  original file or web URL
 //
-case class TextSource(line: Int, col: Int, url: Option[URL] = None) {
+case class TextSource(line: Int, col: Int, endLine: Int, endCol: Int) extends Ordered[TextSource] {
+
+  override def compare(that: TextSource): Int = {
+    line - that.line
+  }
+
   override def toString: String = {
-    if (url.isDefined) {
-      s"line ${line} col ${col} of ${url}"
-    } else {
-      s"line ${line} col ${col}"
-    }
+    s"${line}:${col}-${endLine}:${endCol}"
+  }
+
+  lazy val lineRange: Range = line until endLine
+
+  lazy val maxCol: Int = Vector(col, endCol).max
+}
+
+object TextSource {
+  lazy val empty: TextSource = TextSource(0, 0, 0, 0)
```

Comment:
This doesn't have to be `lazy`. The lazy keyword is useful for computations that are expensive, take a lot or memory, or require files that do not exist at the beginning of the execution.